ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java

```diff
@@ -217,6 +222,16 @@ public static Avu newMetadata(String attr, String value, String unit) {
         return avu;
     }
 
+    @Override
+    public Avu  setAvuModelKey(Avu avu) {
+        if (avu != null) {
+            final AutoBean<Avu> avuBean = AutoBeanUtils.getAutoBean(avu);
+            avuBean.setTag(AVU_BEAN_TAG_MODEL_KEY, String.valueOf(unique_avu_id++));
+            return avuBean.as();
```

Comment:
This may not be correct, but I'm a little confused since this looks like the code that's in the ModelKeyProvider in the `createAdditionalListStore` method.  It seems like at least some of the time you're using `setAvuModelKey` right before adding to the liststore.  In that case, I wonder if it makes sense to modify `createAdditionalListStore` to return the key/tag if it exists, and then if it doesn't exist to create/set it and then return it.  This could just be a preference type of change though.